crates/router/src/core/payment_methods.rs

```diff
@@ -44,7 +44,10 @@ pub trait PaymentMethodRetrieve {
         payment_intent: &PaymentIntent,
         card_token_data: Option<&CardToken>,
         customer: &Option<domain::Customer>,
-    ) -> RouterResult<Option<(payments::PaymentMethodData, enums::PaymentMethod)>>;
+    ) -> (
+        RouterResult<Option<(payments::PaymentMethodData, enums::PaymentMethod)>>,
+        Option<String>,
+    );
```

Comment:
`RouterResult` should encapsulate the entire return type. You can also consider creating a struct for the output, since it's becoming complex. You can put the struct definition in `router::types::storage::payment_method`.

```suggestion
    ) -> RouterResult<(Option<(payments::PaymentMethodData, enums::PaymentMethod)>, Option<String>)>;
```